### PR TITLE
Fix for ODEs with no parameters

### PR DESCRIPTION
**Pull request by sumiya11**
I have deja vu, didn't we fix this before 😅  ?
@pogudingleb 

Anyways, this PR hopefully fixes it for good.

```julia
using StructuralIdentifiability
ode = @ODEmodel(
           x1'(t) = x1,
           x2'(t) = x2,
           y(t) = x1 + x2
       )

assess_identifiability(ode)
[ Info: Assessing local identifiability
[ Info: Local identifiability assessed in 0.1328883 seconds
[ Info: Assessing global identifiability
[ Info: Computing IO-equations
┌ Info: Computed in 0.1024635 seconds
│   :ioeq_time = :ioeq_time
└   ioeq_time = 0.1024635
[ Info: No parameters, so Wronskian computation is not needed
ERROR: need at least one indeterminate
Stacktrace:
  [1] error(s::String)
    @ Base .\error.jl:35
  [2] #29
    @ C:\Users\User\.julia\packages\Nemo\9YO6X\src\flint\FlintTypes.jl:1466 
[inlined]
  [3] (::AbstractAlgebra.var"#51#52"{Nemo.var"#29#30"{Vector{Symbol}, Symbol}, AbstractAlgebra.WeakValueDict{Tuple{Vector{Symbol}, Symbol}, Nemo.QQMPolyRing}, Tuple{Vector{Symbol}, Symbol}})()
    @ AbstractAlgebra C:\Users\User\.julia\packages\AbstractAlgebra\kV4qH\src\WeakValueDict.jl:554
...
```

### CODE REVIEW

**Comment by pogudingleb** [comment]
Weird indeed! I vividly remember fixing this... Maybe I have merged it somewhere else.

Is it intentional that you have `x1` as a states (with `(t)`) and as a parameter (without `(t)`) in the tests?

**Comment by sumiya11** [comment]
> Is it intentional that you have x1 as a states (with (t)) and as a parameter (without (t)) in the tests?

No, not really. Though, I think `x1` is declared as a state even without `(t)`?

```julia
using StructuralIdentifiability
ode = @ODEmodel(x1'(t) = x1, x2'(t) = x2, y(t) = x1 + x2)
[ Info: Summary of the model:
[ Info: State variables: x1, x2
[ Info: Parameters:
[ Info: Inputs:
[ Info: Outputs: y
```

**Comment by pogudingleb** [comment]
Indeed!
I wonder if this is a bug or a feature, looks more like feature...

**Comment by pogudingleb** [comment]
@sumiya11  Thanks again for fixing this!
I have changed the line `states_needed = states_needed || isempty(ode.parameters)` to something which occurred to be more intuitive for me. If this looks fine, feel free to merge.

File: src/global_identifiability.jl [hunk 1]
```diff
@@ -176,7 +205,9 @@ function check_identifiability(
             break
         end
     end
-    states_needed = states_needed || isempty(ode.parameters)
+    if !states_needed && isempty(ode.parameters)
+        return [true for _ in funcs_to_check]
```
Comment (sumiya11) [inline src/global_identifiability.jl:209]: Isn't this vector always empty?

Comment (pogudingleb) [reply src/global_identifiability.jl:209]: In a meaningful situation, no, but technically one can write things like `assess_identifiability(ode, funcs_to_check=[one(ode.poly_ring)])`

**Pull request merged**